TechAndCheck/zenodotus
# Restyle the display of media elements and archive items

This PR delivers the bulk of #219, restyling the display of archive items on the index page.

**Before:**
<img width="794" alt="Screenshot of an archive item, before redesign" src="https://user-images.githubusercontent.com/4731/171065489-f80b7d9c-99f9-4825-9599-f753f12c4eb6.png">

**After:**
<img width="777" alt="Screenshot of an archive item, after redesign" src="https://user-images.githubusercontent.com/4731/171934119-bbc45d1a-e174-494d-bd9f-536a6aa384b2.png">

**Notes:**
- These features were determined to be just out of scope, and will be done as separate work: #225.
- So far, this styling only applies on the archive index page. Applying it to other pages has been bumped to #224.

**Testing:**
- `yarn install` to get a new package (Day.js)
- `./bin/dev`
- Log in and go to the root of the site ("Most Recent Archived Media") and observe the new item design. Especially:
   - Scrape some multi-image tweets or Instagram posts and make sure they display as expected
   - Click author headers to go to author pages
   - Pay attention to "Published" and "Archived" timestamps, which should appear in your local time

**Feedback wanted:** Obviously, any problems you find, but also **your thoughts/criticisms of the design** and **any concerns you have with implementation**. In particular, I made heavy use of the `ArchivableItem` module, and even (😬) exposed the Rails URL helpers to the models (see `author_canonical_path` in `normalized_attrs_for_views` in any source model). Please let me know if you're uncomfortable with any of this!

Closes #219 
Closes #222
Closes #226
